NWN.Permissions/src/ChatCommands/UserAddPermissionCommand.cs

```diff
@@ -0,0 +1,66 @@
+using System;
+using System.Collections.Generic;
+using Anvil.API;
+using Anvil.Services;
+using Jorteck.ChatTools;
+
+namespace Jorteck.Permissions
+{
+  [ServiceBinding(typeof(IChatCommand))]
+  internal class UserAddPermissionCommand : IChatCommand
+  {
+    private readonly ConfigService configService;
+
+    public string Command => configService.GetFullChatCommand("user addpermission");
+    public string[] Aliases => null;
+
+    public Dictionary<string, object> UserData { get; } = new Dictionary<string, object>
+    {
+      [PermissionConstants.ChatUserDataKey] = PermissionConstants.UserAddPermission,
+    };
+
+    public int? ArgCount => 1;
+    public string Description => "Grants a permission to a user.";
+
+    public CommandUsage[] Usages { get; } =
+    {
+      new CommandUsage("<permission_name>", "Grant the specified permission to the target user."),
+    };
+
+    public UserAddPermissionCommand(ConfigService configService)
+    {
+      this.configService = configService;
+    }
+
+    public void ProcessCommand(NwPlayer caller, IReadOnlyList<string> args)
+    {
+      string permission = args[0];
+      caller.EnterTargetMode(eventData => eventData.ProcessOnValidPCTargetBy(caller, AddUserPermissionToTarget(permission)));
+    }
+
+    private Action<NwPlayer, NwPlayer> AddUserPermissionToTarget(string permission)
+    {
+      return (NwPlayer target, NwPlayer caller) =>
+      {
+        configService.UpdateUserConfig(config =>
+        {
+          if (!config.UsersCd.TryGetValue(target.CDKey, out UserEntry entry))
+          {
+            entry = new UserEntry();
+            config.UsersCd[target.CDKey] = entry;
+          }
```

Comment:
What are your thoughts on the permission commands always operating on player CD keys? Should it be customizable to work on player names and UUIDs too? (per-character vs per account)

I know I hard coded it in the `group add` command.

Reply:
I was wondering if you had any thoughts about the same. As you can tell I simply copied the logic from `group add`.

Since the plugin is generic it makes sense to support permissions on the character level, but I found it tricky to define the use cases. Is it OK to progress with CD Key only for now and expand the functionality later?